plugin-tray/sniproxy.cpp

```diff
@@ -580,20 +582,23 @@ void SNIProxy::sendClick(uint8_t mouseButton, int x, int y)
         else
             configVals[1] = static_cast<uint32_t>(y - clickPoint.y());
     }
-    xcb_configure_window(c, m_containerWid, XCB_CONFIG_WINDOW_X | XCB_CONFIG_WINDOW_Y, configVals);
+    xcb_configure_window(m_connection, m_containerWid, XCB_CONFIG_WINDOW_X | XCB_CONFIG_WINDOW_Y, configVals);
 
     // pull window up
     stackContainerWindow(XCB_STACK_MODE_ABOVE);
 
+    auto *x11Application = qGuiApp->nativeInterface<QNativeInterface::QX11Application>();
+    WId appRootWindow = XDefaultRootWindow(x11Application->display());
+
     // mouse down
     if (m_injectMode == Direct) {
         xcb_button_press_event_t *event = new xcb_button_press_event_t;
         memset(event, 0x00, sizeof(xcb_button_press_event_t));
         event->response_type = XCB_BUTTON_PRESS;
         event->event = m_windowId;
-        event->time = QX11Info::getTimestamp();
+        event->time = XCB_CURRENT_TIME; //FIXME: how to get proper timestamp?
```

Comment:
No need for comment. Setting XCB_CURRENT_TIME should be fine.

Reply:
In following PRs  I've linked to Qt private API which was needed for KDE Window managment protocol requiring wl_surface.
So I've also changed this to use private platform integration resources